### ISSUE TITLE
Manage default toolboxes with systemd/podman

### ISSUE THREAD

**Issue by castrojo** [ublue-os/bluefin#736]
https://gitlab.gnome.org/chergert/prompt/-/issues/31

Summary from chat, we need to just manage the default toolboxes we ship with podman and systemd. That way we can label the containers properly and use the podman auto update service and all the goodies. This is probably how we should have set them up in the first place. 

This will effectively remove distrobox from the creation and entering of the default containers, which will let us slim them down more. 

Then distrobox is just used for all the user-created containers so those workflows will still work great.

**Comment by johnnylam88** [comment]
I don't think this can be *bluefin* or *-dx* only? This pops up on *bazzite* as well.

**Comment by castrojo** [comment]
Agree 100%, perhaps a common set of quadlets for a default "system toolbox" might be a clean solution and once we figure it out anyone can set up their linux this way.

**Comment by m2Giles** [comment]
Is the goal to not have these managed at all by distrobox/toolbox? Instead quadlet and prompt for access?

Might make some things like distrobox-host-exec/distrobox-enter more difficult

**Comment by castrojo** [comment]
Yeah that's why I wanted us to talk it through. Plusses and minuses to each approach, etc.

"I hope we don't have to copy and paste `distrobox-create` into a service unit" has for sure come up, haha.

**Comment by m2Giles** [comment]
I think the bluefin-cli container would be a good candidate for testing this out. Or the wolfi image. They are both aimed at terminal applications.

Bluefin-cli already has host-spawn and distrobox-host-exec built into it. But needs an  environment variable to work.

After I poke at some Nvidia stuff I'm looking back here.

**Comment by castrojo** [comment]
oh yes good point, at least for wolfi and bluefin-cli, there are no GUI apps, you wouldn't need all the X stuff, etc.

**Comment by m2Giles** [comment]
Example quadlets and systemd one-shots are available in ublue-os/toolboxes.

Quadlets can be used for distrobox and toolbox containers. They will be recreated on each login. We cannot control enable since they generate a transient unit. We could possibly have a custom systems target to control enabling them.

That may be a bit aggressive as a default so oneshots that leverage distrobox assemble files is another option. This is also easier for an individual to extend because they could just modify an assemble file instead of making an image like the quadlet (distrobox quadlet can modify additional packages but not as accessible as assemble file).

For shipped enter a fedora toolbox or Ubuntu toolbox, I can see argument for using quadlet. Clean slate each time you login. Auto image updates if available.

I think the main is having these things on image but not enabled (just a few ini style files). And having ujust commands for configuring cli as part of motd/welcome.

**Comment by castrojo** [comment]
> And having ujust commands for configuring cli as part of motd/welcome.

Cross linking: https://github.com/ublue-os/bluefin/issues/609

**Comment by m2Giles** [comment]
Status for this.

Right now there are quadlets for:
1. Bluefin-cli
2. Bluefin-dx-cli
3. Fedora-toolbox
4. Ubuntu-toolbox
5. Wolfi-toolbox 
6. Wolfi-dx-toolbox

These are on disk. These can be enabled for auto start by enabling a user systems target of the same name.

When these then run, a script will create a profile in prompt to enter into the container. There is a convenience script for bluefin-cli right now to enable and set it as default. This is accessible via `ujust bluefin-cli`

There is a wip script for all other containers under `ujust configure-terminal`. This script will look more like the bluefin-cli script.

**Comment by m2Giles** [comment]
Quadlets now used for bluefin-cli, Ubuntu and fedora. Shortcuts working on bluefin and Aurora 

Auto updates handled via podman update timer.
